WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -378,6 +381,47 @@ public static PostModel updatePostWithCurrentlyFailedUploads(PostModel post) {
         return post;
     }
 
+    // this method aims at recovering the current state of media items if they're inconsistent within the PostModel.
+    public static void resetUploadingMediaToFailedIfNotInProgressOrQueued(Context context, PostModel post) {
```

Comment:
:man_facepalming:  you're so much right, sorry - moved to `EditPostActivity` where it truly belongs, as within this method we also make use of `UploadService` and we can't reference that one from within `AztecEditorFragment`. In 096290b